stl/inc/optional

```diff
@@ -245,22 +252,25 @@ public:
 
     template <class _Ty2,
         enable_if_t<conjunction_v<_AllowUnwrapping<_Ty2>, is_constructible<_Ty, const _Ty2&>>, int> = 0>
-    _CONSTEXPR20 explicit(!is_convertible_v<const _Ty2&, _Ty>) optional(const optional<_Ty2>& _Right) {
+    _CONSTEXPR20 explicit(!is_convertible_v<const _Ty2&, _Ty>)
+        optional(const optional<_Ty2>& _Right) noexcept(is_nothrow_constructible_v<_Ty, const _Ty2&>) {
         if (_Right) {
             this->_Construct(*_Right);
         }
     }
 
     template <class _Ty2, enable_if_t<conjunction_v<_AllowUnwrapping<_Ty2>, is_constructible<_Ty, _Ty2>>, int> = 0>
-    _CONSTEXPR20 explicit(!is_convertible_v<_Ty2, _Ty>) optional(optional<_Ty2>&& _Right) {
+    _CONSTEXPR20 explicit(!is_convertible_v<_Ty2, _Ty>)
+        optional(optional<_Ty2>&& _Right) noexcept(is_nothrow_constructible_v<_Ty, _Ty2>) /* strengthened */ {
         if (_Right) {
             this->_Construct(_STD move(*_Right));
         }
     }
 
 #if _HAS_CXX23
     template <class _Fn, class _Ux>
-    constexpr optional(_Construct_from_invoke_result_tag _Tag, _Fn&& _Func, _Ux&& _Arg)
+    constexpr optional(_Construct_from_invoke_result_tag _Tag, _Fn&& _Func, _Ux&& _Arg) noexcept(
+        is_nothrow_constructible_v<_Mybase, _Construct_from_invoke_result_tag, _Fn, _Ux>)

```

Comment:
`_Optional_destruct_base(_Construct_from_invoke_result_tag, _Fn&& _Func, _Ux&& _Arg)` wasn't updated, so this is always `noexcept(false)`.